framework/src/org/checkerframework/common/signatureinference/SignatureInferenceScenes.java

```diff
@@ -0,0 +1,726 @@
+
+    /** Maps .jaif file paths (Strings) to Scenes. */
+    private static Map<String, AScene> scenes = new HashMap<>();
+
+    /**
+     * Set representing Scenes that were modified since the last time all
+     * Scenes were written into .jaif files. Each String element of this set
+     * is a path to the .jaif file of the corresponding Scene in the set. It
+     * is obtained by passing a class name as argument to the
+     * {@link #getJaifPath} method.
+     * <p>
+     * Modifying a Scene means adding (or changing) a type annotation for a
+     * field, method return type, or method parameter type in the Scene.
+     * (Scenes are modified by the method {@link #updateAnnotationSetInScene}.)
+     */
+    private static Set<String> modifiedScenes = new HashSet<>();
+
+    /**
+     * Returns the String representing the .jaif path of a class given its name.
+     */
+    private static String getJaifPath(String className) {
+        String jaifPath = JAIF_FILES_PATH + className + ".jaif";
+        return jaifPath;
+    }
+
+    /**
+     * Returns the Scene stored in a .jaif file path passed as input.
+     * If the file does not exist, an empty Scene is created.
+     */
+    private static AScene getScene(String jaifPath) {
+        if (!scenes.containsKey(jaifPath)) {
+            File jaifFile = new File(jaifPath);
+            AScene scene = new AScene();
+            if (jaifFile.exists()) {
+                try {
+                    IndexFileParser.parseFile(jaifPath, scene);
+                } catch (IOException e) {
+                    ErrorReporter.errorAbort("Problem while reading file in: " + jaifPath + "."
+                            + " Exception message: " + e.getMessage());
+                }
+            }
+            scenes.put(jaifPath, scene);
+        }
+        return scenes.get(jaifPath);
+    }
+
+    /**
+     * Write all modified scenes into .jaif files.
+     * (Scenes are modified by the method {@link #updateAnnotationSetInScene}.)
+     */
+    public static void writeScenesToJaif() {
+        // Create .jaif files directory if it doesn't exist already.
+        File jaifDir = new File(JAIF_FILES_PATH);
+        if (!jaifDir.exists()) {
+            jaifDir.mkdirs();
+        }
+        // Write scenes into .jaif files.
+        for (String jaifPath : modifiedScenes) {
+            try {
+                AScene scene = scenes.get(jaifPath);
+                IndexFileWriter.write(scene, new FileWriter(jaifPath));
+            } catch (IOException e) {
+                ErrorReporter.errorAbort("Problem while reading file in: " + jaifPath
+                        + ". Exception message: " + e.getMessage());
+            } catch (DefException e) {
+                ErrorReporter.errorAbort(e.getMessage());
+            }
+        }
+        modifiedScenes.clear();
+    }
+
+    /**
+     * Updates the parameter types of the method methodElt in the Scene of the
+     * class with symbol classSymbol.
```

Comment:
The Javadoc is strange because it refers to a local variable classSymbol. Someone reading the Javadoc by itself would not see the declaration of classSymbol. Would it be better to describe it in terms of receiverTree?